Lib/test/test_capi/test_opt.py

```diff
@@ -2219,6 +2236,13 @@ def f(n):
         self.assertNotIn("_LOAD_ATTR_METHOD_NO_DICT", uops)
         self.assertNotIn("_LOAD_ATTR_METHOD_LAZY_DICT", uops)
 
+class TestObject:
+    def test(self, *args, **kwargs):
+        return args[0]
+
+test_object = TestObject()
+test_bound_method = TestObject.test.__get__(test_object)
+
```

Comment:
Can you move this into `test_method_guards_removed_or_reduced`? `global_identity` needs to be at module scope for other reasons that don't apply here.

Also, I think it can be simplified to something like:

```py
class TestObject:
    def test(self, arg):
        return arg

test_bound_method = TestObject().test
```

Reply:
If I'm correct, `TestObject().test` will just generate `_CHECK_FUNCTION_VERSION` code, It's a normal function. 

Reference the test code here

Reply:
> Can you move this into `test_method_guards_removed_or_reduced`? `global_identity` needs to be at module scope for other reasons that don't apply here.

BTW, if I push the test object into the `test_method_guards_removed_or_reduced`. I think the method will not be treated by a const. So we the `_CHECK_FUNCTION_VERSION_INLINE` is not exist in final opcode.

Reply:
> Can you please move this back to global scope? It seems tests are failing as the global method is not being promoted to a constant.

Sure